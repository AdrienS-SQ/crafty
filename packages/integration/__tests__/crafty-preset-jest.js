/* global describe, it, expect */

const path = require("path");

const rimraf = require("rimraf");

const testUtils = require("../utils");

let testIfNotPnp = it;

try {
  require("pnpapi");
  testIfNotPnp = it.skip;
} catch (error) {
  // not in PnP; not a problem
}

it("Succeeds without transpiling", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-jest/succeeds")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["test"]);

  expect(result).toMatchSnapshot();
});

it("Creates IDE Integration files", () => {
  process.chdir(path.join(__dirname, "../fixtures/crafty-preset-jest/ide"));
  rimraf.sync("jest.config.js");
  rimraf.sync(".gitignore");

  const result = testUtils.run(["ide"]);

  expect(result).toMatchSnapshot();
  expect(testUtils.readForSnapshot("jest.config.js")).toMatchSnapshot();

  expect(testUtils.readForSnapshot(".gitignore")).toMatchSnapshot();
});

it("Creates IDE Integration files with Babel", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-jest/ide-babel")
  );
  rimraf.sync("jest.config.js");
  rimraf.sync(".gitignore");

  const result = testUtils.run(["ide"]);

  expect(result).toMatchSnapshot();
  expect(testUtils.readForSnapshot("jest.config.js")).toMatchSnapshot();
});

testIfNotPnp("Succeeds with typescript", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-jest/typescript")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["test"]);

  expect(result).toMatchSnapshot();
});

it("Succeeds with babel", () => {
  process.chdir(path.join(__dirname, "../fixtures/crafty-preset-jest/babel"));
  rimraf.sync("dist");

  const result = testUtils.run(["test"]);

  expect(result).toMatchSnapshot();
});

testIfNotPnp("Succeeds with babel and React", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-jest/babel-react")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["test"]);

  expect(result).toMatchSnapshot();
});

it("Succeeds with esm module", () => {
  process.chdir(path.join(__dirname, "../fixtures/crafty-preset-jest/esm"));
  rimraf.sync("dist");

  const result = testUtils.run(["test"]);

  expect(result).toMatchSnapshot();
});

it("Succeeds with esm module and babel", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-jest/esm-babel")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["test"]);

  expect(result).toMatchSnapshot();
});
