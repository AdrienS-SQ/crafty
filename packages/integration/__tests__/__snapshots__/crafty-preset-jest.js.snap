// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Succeeds with babel 1`] = `
Object {
  "status": 0,
  "stderr": "PASS src/__tests__/math.js
  ✓ adds two numbers (__ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
",
}
`;

exports[`Succeeds with typescript 1`] = `
Object {
  "status": 0,
  "stderr": "PASS src/__tests__/math.ts
  ✓ adds two numbers (__ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
",
}
`;

exports[`Succeeds without transpiling 1`] = `
Object {
  "status": 0,
  "stderr": "PASS src/__tests__/math.js
  ✓ adds two numbers (__ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
",
}
`;
